src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java

```diff
@@ -115,10 +119,23 @@ public NettyPgProtocolStream(EventLoopGroup group, SocketAddress address, boolea
 
             pushSubscriber(subscriber);
             write(messages);
+            initReadTimeout(subscriber);
 
         }).lift(throwErrorResponsesOnComplete());
     }
 
+    private void initReadTimeout(Subscriber subscriber) {
+        if (readTimeout != 0) {
+            Runnable onTimeout = () -> {
+                ReadTimeoutException timeoutException = ReadTimeoutException.INSTANCE;
+                subscriber.onError(timeoutException);
```

Comment:
Won't this lead to duplicate onError calls? I would assume fireExceptionCaught would propagate the exception to all subscribers.

Reply:
initially it was wired like that. but for some reason I could not see this exception bubble up on exception handler.